lib/components/user/nav-login-button.js

```diff
@@ -0,0 +1,124 @@
+      target: PropTypes.string,
+      text: PropTypes.string.isRequired,
+      url: PropTypes.string.isRequired
+    })),
+    onSignInClick: PropTypes.func.isRequired,
+    onSignOutClick: PropTypes.func.isRequired,
+    profile: PropTypes.shape({
+      email: PropTypes.string.isRequired,
+      name: PropTypes.string.isRequired,
+      nickname: PropTypes.string,
+      picture: PropTypes.string
+    })
+  }
+
+  static defaultProps = {
+    links: null,
+    profile: null
+  }
+
+  render () {
+    const {
+      className,
+      id,
+      links,
+      onSignInClick,
+      onSignOutClick,
+      profile,
+      style
+    } = this.props
+
+    const commonProps = {
+      className,
+      id,
+      style
+    }
+
+    /**
+      * Override bootstrap's defaults, especially for nav links.
+      * HACK: Because we don't want to use the rendered <NavDropDown> inside a <Nav> while in mobile mode
+      * (otherwise, the menu items get displayed full screen width, which is undesirable),
+      * the CSS rules below let us fit the sign-in component to the top-right corner
+      * and add some of the default visuals provided by react-bootstrap's <Nav> while in mobile mode.
+      * We change the link color to white for consistency with the AppMenu component.
+      */
+    const styleOverrides = (
+      <style>{`
+        .otp.mobile .navbar .container-fluid > li {
+          display: block;
+          padding: 14px;
+          position: fixed;
+          right: 0;
+          top: 0;
+        }
+        .otp.mobile .navbar li.dropdown.open {
+          background-color: #080808;
+        }
+        .navbar-inverse .navbar-nav > li > a,
+        .otp.mobile .navbar .container-fluid > li > a,
+        .otp.mobile .navbar .container-fluid > li.dropdown.open > a {
+          color: #fff;
+          text-decoration: none;
+        }
+        .navbar-inverse .navbar-nav > li > a:hover,
+        .otp.mobile .navbar .container-fluid > li > a:hover {
+          color: #ddd;
+          text-decoration: none;
+        }
+      `}
+      </style>
+    )
+
+    if (profile) {
+      const displayedName = profile.nickname || profile.name
+      return (
+        <NavDropdown
+          {...commonProps}
+          pullRight
+          title={<span>
+            {styleOverrides}
+            <img
+              alt={displayedName}
+              src={profile.picture}
+              style={{width: '2em', height: '2em', margin: '-15px 0'}}
+              title={`${displayedName}\n(${profile.email})`}
+            />
+          </span>
+          }>
+          <MenuItem header>{displayedName}</MenuItem>
+
+          {links && links.map((item, index) => (
+            <MenuItem href={item.url} key={index} target={item.target}>{item.text}</MenuItem>
+          ))}
+          <MenuItem divider />
+          <MenuItem onSelect={onSignOutClick}>Sign out</MenuItem>
+        </NavDropdown>
+      )
+    }
+
+    return (
```

Comment:
Could you give some basic comments next to the above conditional and this return statement about the presumed login states and what we're returning? I.e.:

```
User is logged in, return dropdown with links to account information and log out.
```

and 

```
User not logged in, showing sign in link.
```